### PR TITLE
Refactors some procs from unsorted.dm helper to better places

### PR DESCRIPTION
**Pull request by Amunak**
Title. Additionally, "dropsafety" for checking borg modules is no longer a global proc - instead it's called directly on atoms. Oh and fixed some `proc/thing` to `/proc/thing`.

Definitely more could've been done but I didn't want to overdo it, so I just (kind of arbitrarily) moved the worst offenders, going mostly by type - if it's a proc on type, I moved it to the file for the type, and besides `dropsafety` I didn't actually change any logic, just moved it around. `dropsafety` now additionally checks if `loc` is a thing.

### CODE REVIEW

**Comment by Amunak** [comment]
!review

File: code/game/atoms.dm [hunk 1]
```diff
@@ -109,6 +116,86 @@
...
 
+// Helper for adding verbs with timers.
+/atom/proc/add_verb(the_verb, datum/callback/callback)
+	if (callback && !callback.Invoke())
+		return
+
+	verbs += the_verb
+
+#define NOT_FLAG(flag) (!(flag & use_flags))
+#define HAS_FLAG(flag) (flag & use_flags)
```
Comment (mikomyazaki) [inline code/game/atoms.dm:127]:
```suggestion
#define HAS_FLAG(flag) (flag & use_flags)
#define NOT_FLAG(flag) !HAS_FLAG(flag)
```

File: code/game/atoms.dm [hunk 2]
```diff
@@ -109,6 +116,86 @@
...
+
+// Checks if user can use this object. Set use_flags to customize what checks are done.
+// Returns 0 if they can use it, a value representing why they can't if not.
+// Flags are in `code/__defines/misc.dm`
+/atom/proc/use_check(mob/user, use_flags = 0, show_messages = FALSE)
+	. = 0
```
Comment (mikomyazaki) [inline code/game/atoms.dm:133]:
```suggestion
	. = USE_SUCCESS
```

File: code/game/atoms.dm [hunk 3]
```diff
@@ -109,6 +116,86 @@
...
+
+#undef NOT_FLAG
+#undef HAS_FLAG
+
+/atom/proc/get_light_and_color(var/atom/origin)
+	if(origin)
+		color = origin.color
+		set_light(origin.light_range, origin.light_power, origin.light_color)
+
+/atom/proc/find_up_hierarchy(var/atom/target)
```
Comment (mikomyazaki) [inline code/game/atoms.dm:179]: Do we need this *and* `/proc/get()` they seem really similar?

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

File: code/game/atoms.dm (hunk 3 above)
Comment (Amunak) [reply code/game/atoms.dm:179]: `find_up_hierarchy` is used in just one place, to check validity of modifier items by checking if it's still in the inventory (or same loc I guess). But it checks a specific `/atom` while `get()` checks just a type, so technically they are different (though we could make one do both if we really wanted).

I'd rather not touch it though, better not break it, lol.

**Review by Geevies — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Review by mikomyazaki — APPROVED** [review]

**Pull request merged**